### PR TITLE
Include cancelled requests and appointments on upcoming and requested pages when using VAOS service

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
This PR
- Includes cancelled requests and appointments on the same pages they're on when not using the vaos service
- Fixes a flaky test

## Testing done
Local and unit testing

## Acceptance criteria
- [ ] Cancelled requests appear correctly when using vaos service

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/services/appointment/index.js [hunk 1]
```diff
@@ -125,9 +125,14 @@ export async function getBookedAppointments({
     if (useV2) {
       const appointments = await getAppointments(startDate, endDate, [
         'booked',
+        'cancelled',
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:128]: The changes in this file are the core changes

File: src/applications/vaos/services/appointment/index.js [hunk 2]
```diff
@@ -125,9 +125,14 @@ export async function getBookedAppointments({
...
       ]);
 
-      return transformVAOSAppointments(appointments);
+      const appointmentsWithoutRequests = appointments.filter(
+        appt => !appt.requestedPeriods,
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/index.js:132]: Items with a cancelled status could either be cancelled requests or cancelled appointments, so we need to filter out the ones we don't want depending on the function we're in.

File: src/applications/vaos/services/appointment/transformers.v2.js [hunk 3]
```diff
@@ -46,6 +48,10 @@ function getTypeOfVisit(id) {
 export function transformVAOSAppointment(appt) {
   const isCC = appt.kind === 'cc';
   const isVideo = appt.kind === 'telehealth';
+  const vistaId = appt.locationId?.substr(0, 3);
+  const timezone = getTimezoneBySystemId(vistaId)?.timezone;
+
+  const start = timezone ? moment(appt.start).tz(timezone) : moment(appt.start);
```
Comment (jbalboni) [inline src/applications/vaos/services/appointment/transformers.v2.js:54]: @jenniemc's PR has similar changes, so I'll remove these changes if that gets merged first.

File: src/applications/vaos/tests/mocks/v2.js [hunk 4]
```diff
@@ -41,12 +40,48 @@ export function getVAOSRequestMock() {
...
   };
 }
 
+/**
+ * Return a stubbed VAOS VA appointment item
+ *
+ * @export
+ * @returns {VAOSRequest} VAOS VA request object
+ */
+export function getVAOSAppointmentMock() {
```
Comment (jbalboni) [inline src/applications/vaos/tests/mocks/v2.js:55]: Needed to create a separate default mock function for requests and appointments, even though the differences are pretty subtle.

**Review by LeKeve — APPROVED** [review]

**Review by vbahinwillit — COMMENTED** [review]
LGTM

**Pull request merged**